src/pet.js

```diff
@@ -0,0 +1,125 @@
+//magic numbers
+const MAX_FITNESS = 10;
+const MIN_HUNGER = 0;
+const HUNGRY = "I am hungry"
+const UNFIT = "I need a walk"
+const BOTH = "I am hungry AND I need a walk"
+const FINE = "I feel great!"
+const DIED = "Your pet is no longer alive :("
+const ERROR = new Error('Your pet is no longer alive :(')
+
+function Pet(name) {
+    this.name = name;
+    this.age = 0;
+    this.hunger = 0;
+    this.fitness = 4;
+    this.children=[]
+  
+
+}
+
+//getter function 
+Pet.prototype = {
+    get isAlive() {
+        return this.age < 30 && this.hunger < 10 && this.fitness > 0;
+    }
+}
+
+
+
+Pet.prototype.growUp = function () {
+    this.age += 1;
+    this.hunger += 5;
+    this.fitness += 3;
+}
+
+Pet.prototype.walk = function () {
+    if (this.fitness + 4 <= MAX_FITNESS) {
+        this.fitness += 4;
+    }
+    else {
+        this.fitness = MAX_FITNESS;
+    }
+}
+
+Pet.prototype.feed = function () {
+    if (!this.isAlive) {
+        throw ERROR;
+    }
+
+    if (this.hunger < 0) {
+        this.hunger = MIN_HUNGER;
+    }
+
+    if (this.hunger - 3 >= MIN_HUNGER) {
+        this.hunger -= 3;
+    }
+
+
+}
+
+Pet.prototype.checkUp = function () {
+    if (this.hunger >= 5 && this.fitness <= 3) {
+        return BOTH
+    }
+    if (this.hunger < 5 && this.fitness > 3) {
+        return FINE
+    }
+    if (this.fitness <= 3) {
+        return UNFIT
+    }
+    if (this.hunger >= 5) {
+        return HUNGRY
+    }
+
+}
+
+
+//add method to pet constructor 
+Pet.prototype.haveBaby=function(babyname){
+    const child = new Pet (babyname)
+    return child.babyname;
+    
+ }
```

Comment:
this doesn't work. You are creating a `new Pet` with the name the function is called with, but you aren't assigning this value anywhere so it is lost from memory. You are then returning `child.babyname`, but `babyname` is not a property of a `pet` instance, so this function returns `undefined`. The functionality for this function should look like:
```
Pet.prototype.haveBaby = function (name) {
  const child = new Pet(name);
  this.children.push(child);
  return child;
}
```
this way, you create the baby like you did before, but you add the instance of the baby to the `children` array on the parent. Because you've added the whole instance, you can then access the babies properties through the parent, e.g.
```
const parent = new Pet("Fido");
parent.haveBaby("Jimmy");
parent.children[0].checkUp(); //will call the checkUp method on the first child of the parent
```
I have also returned the baby so that you could assign it to a variable if you prefer, e.g.
```
const parent = new Pet("Fido");
const child = parent.haveBaby("Jimmy");
child.checkUp();
``

Reply:
If you wanted to add the `adoptChild` functionality, that would look like this:
```
Pet.prototype.adoptChild = function(child) {
  this.children.push(child);
}
```
to use this function, you would do something like this:
```
const parent = new Pet("Fido");
const child = new Pet("Jimmy");
parent.adoptChild(child);
```